ext/handlers/conversation.go

```diff
@@ -36,14 +36,14 @@ type Conversation struct {
 }
 
 type ConversationOpts struct {
+	// StateStorage is responsible for storing all running conversations.
+	StateStorage conversation.Storage
 	// Exits is the list of handlers to exit the current conversation partway (eg /cancel commands)
 	Exits []ext.Handler
 	// Fallbacks is the list of handlers to handle updates which haven't been matched by any states.
 	Fallbacks []ext.Handler
 	// If True, a user can restart the conversation by hitting one of the entry points.
 	AllowReEntry bool
-	// StateStorage is responsible for storing all running conversations.
-	StateStorage conversation.Storage
```

Comment:
So, you mention that:
> For example, handlers.Conversation became a consolidated 48 bytes of pointers instead of 72 bytes. If we follow 10 000 conversations this may well reduce the GC interruption time.

By "follow 10 000 conversations", do you mean having a conversation with 10 000 users? Is GC interruption time an issue for anyone at the moment...?

Because this struct is only used when the conversation itself is created; eg, in this case, it is used exactly as many times as `dispatcher.AddHandler(handlers.NewConversation(...))` is called. Which, for what I expect to be the VAST majority of cases, would only be once, at startup. Given the nature of this libary, things are going to be primarily network IO bound, so optimising a one-time call seems a little overkill.

I'm happy to merge the PR, but I'm just wary of these calculations which seem to be suspiciously close to premature optimisation based on a misunderstanding of the struct function 😅